Framework/DataHandling/src/LoadNXcanSAS.cpp

```diff
@@ -283,10 +278,14 @@ void read2DWorkspace(H5::DataSet &dataSet,
   hsize_t memSpaceDimension[1] = {dimInfo.dimBin};
   H5::DataSpace memSpace(1, memSpaceDimension);
 
+  auto &dataHist = func(workspace, 0);
+  Mantid::MantidVec data(dataHist.size());
   for (size_t index = 0; index < dimInfo.dimSpectrumAxis; ++index) {
     // Set the dataSpace to a 1D HyperSlab
     fileSpace.selectHyperslab(H5S_SELECT_SET, sizeOfSingleSlab, start);
-    dataSet.read(func(workspace, index), memoryDataType, memSpace, fileSpace);
```

Comment:
No, `func(workspace, index)` was returning a pointer to a double array. Now the double array is passed directly. `func(workspace, index)` now returns a `HistogramData` type, so we just copy the data.